source/gems/dry-transaction/usage.html.md

```diff
@@ -129,18 +145,60 @@ NOTIFICATIONS
 
 This pub/sub mechanism is provided by the [Wisper](https://github.com/krisleech/wisper) gem. You can subscribe to specific steps using the `#subscribe(step_name: listener)` API, or subscribe to all steps via `#subscribe(listener)`.
 
-### Extending transactions
+### Modifying transactions
 
-You can extend existing transactions by inserting or removing steps. See the [API docs](http://www.rubydoc.info/github/dry-rb/dry-transaction/Dry/Transaction/Sequence) for more information.
+You can modify existing transactions by inserting or removing steps (with the modified transaction returned as a copy). See the [API docs](http://www.rubydoc.info/github/dry-rb/dry-transaction/Dry/Transaction/Sequence) for more information.
 
 ### Working with a larger container
 
 In practice, your container won’t be a trivial collection of generically named operations. You can keep your transaction step names simple by using the `with:` option to provide the identifiers for the operations within your container:
 
 ```ruby
-save_user = Dry.Transaction(container: large_whole_app_container) do
-  map :process, with: "attributes.user"
-  try :validate, with: "validations.user", catch: ValidationFailure
-  tee :persist, with: "persistance.commands.update_user"
+save_user = Dry.Transaction(container: LargeAppContainer) do
+  step :process, with: "processors.process_user"
+  step :validate, with: "validation.validate_user"
+  step :persist, with: "persistance.commands.update_user"
+end
+```
+
+### Providing custom step adapters
+
+You can provide your own step adapters to add custom behaviour to transaction steps. Your step adapters must provide a single `#call(step, *args, input)` method, which shoud return the step’s result wrapped in an `Either` object.
+
+You can provide your step adapter in a few different ways. You can add it to the built-in `StepAdapters` container (a [`dry-container`](http://dry-rb.org/gems/dry-container)) to make it available to all transactions in your codebase:
+
+```ruby
+Dry::Transaction::StepAdapters.register :custom_adapter, MyAdapter.new
+```
+
+Or you can make your own container to extend the built-in one, and then pass that to specific transactions:
+
+```ruby
+class MyStepAdapters < Dry::Transaction::StepAdapters
+  register :custom_adapter, MyAdapter.new
+end
+
+save_user = Dry.Transaction(container: Container, step_adapters: MyStepAdapters) do
+  # ...
+end
+```
+
```

Comment:
@solnic @AMHOL – do these 2 dry-container examples seem fair? (i.e. adding an adapter globally by registering it on the existing container, or subclassing it if you want your own separate container object for your custom adapters)